modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavascriptClosureAngularClientCodegen.java

```diff
@@ -83,6 +86,28 @@ public void processOpts() {
         if (!additionalProperties.containsKey(CodegenConstants.HIDE_GENERATION_TIMESTAMP)) {
             additionalProperties.put(CodegenConstants.HIDE_GENERATION_TIMESTAMP, Boolean.TRUE.toString());
         }
+
+        if (additionalProperties.containsKey(USE_ES6)) {
+            setUseEs6(convertPropertyToBooleanAndWriteBack(USE_ES6));
+        }
+    }
+
+    @Override
+    public void preprocessSwagger(Swagger swagger) {
+        super.preprocessSwagger(swagger);
+
+        if (useEs6) {
+            embeddedTemplateDir = templateDir = "javascript-es6-angular";
```

Comment:
@miguelvaladez may I suggest moving the templates to a subfolder under Javascript-Closure-Angular: "Javascript-Closure-Angular/es6" ?